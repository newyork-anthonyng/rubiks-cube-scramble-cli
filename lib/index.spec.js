const lib = require("./");
const { SCRAMBLE_SPLIT_INDEXES, COLORS, getScramble } = lib;

jest.mock("cube-scrambler", () => {
  return () => ({
    scramble: jest.fn(() => [
      "U",
      "U",
      "U",
      "U",
      "U",
      "U",
      "D",
      "D",
      "D",
      "D",
      "D",
      "D",
      "R",
      "R",
      "R",
      "R",
      "R",
      "R",
      "L",
      "L",
      "L",
      "L",
      "L",
      "L",
    ])
  });
});
jest.mock("chalk", () => {
  const text = textColor => bgColor => jest.fn((text) => {
    return `${text} (${bgColor}, ${textColor})`;
  })
  const black = text("black");
  const white = text("white");

  return {
    bgWhite: { black: black("bgWhite") },
    bgYellow: { black: black("bgYellow")},
    bgBlue: white("bgBlue"),
    bgGreen: { black: black("bgGreen") },
    bgRed: white("bgRed"),
    bgKeyword: jest.fn((color) => ({ black: black(`bg${color}`) }))
  };
})

it("should have correct SCRAMBLE_SPLIT_INDEXES", () => {
  expect(SCRAMBLE_SPLIT_INDEXES).toMatchSnapshot();
});

it("should have correct COLORS", () => {
  expect(COLORS).toMatchSnapshot();
});

it("should call console.log with correct arguments", () => {
  const originalConsoleLog = console.log;
  console.log = jest.fn();

  getScramble();

  expect(console.log).toHaveBeenCalledTimes(1);
  expect(console.log.mock.calls[0]).toMatchSnapshot();
  console.log = originalConsoleLog;
});
