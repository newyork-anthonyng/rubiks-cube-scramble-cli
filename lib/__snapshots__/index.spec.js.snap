// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call console.log with correct arguments 1`] = `
Array [
  "U,U,U,U (bgWhite, black)",
  "U,U,D,D (bgYellow, black)",
  "D,D,D,D (bgBlue, white)",
  "R,R,R,R (bgGreen, black)",
  "R,R,L,L (bgRed, white)",
  "L,L,L,L (bgorange, black)",
]
`;

exports[`should have correct COLORS 1`] = `
Array [
  [MockFunction],
  [MockFunction],
  [MockFunction],
  [MockFunction],
  [MockFunction],
  [MockFunction],
]
`;

exports[`should have correct SCRAMBLE_SPLIT_INDEXES 1`] = `
Array [
  Object {
    "beginning": 0,
    "ending": 4,
  },
  Object {
    "beginning": 4,
    "ending": 8,
  },
  Object {
    "beginning": 8,
    "ending": 12,
  },
  Object {
    "beginning": 12,
    "ending": 16,
  },
  Object {
    "beginning": 16,
    "ending": 20,
  },
  Object {
    "beginning": 20,
    "ending": 24,
  },
]
`;
